### PR TITLE
[TF] Add TensorFlow version for LSTM

### PR DESCRIPTION
**Pull request by floriandonhauser**
This PR adds a TensorFlow implementation for LSTM.
It is based on the previous work of @PHOENIXFURY007 and replaces the PR #1598 .
Ir requires no changes to `RNNModel` since they had been made previously so this model accepts states consisting out of multiple values.

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

**Comment by mli** [comment]
Job d2l-en/PR-1683/1 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1683/

File: chapter_recurrent-modern/lstm.md [hunk 1]
```diff
@@ -205,6 +213,30 @@ def get_lstm_params(vocab_size, num_hiddens, device):
     return params
 ```
 
+```{.python .input}
+#@tab tensorflow
+def get_lstm_params(vocab_size, num_hiddens):
+    num_inputs = num_outputs = vocab_size
+
+    def normal(shape):
+        return tf.Variable(tf.random.normal(shape=shape,stddev=0.01,mean=0,dtype=tf.float32))
```
Comment (terrytangyuan) [inline chapter_recurrent-modern/lstm.md:222]: space after comma, line too long (check the style guide in the repo)

File: chapter_recurrent-modern/lstm.md [hunk 2]
```diff
@@ -205,6 +213,30 @@ def get_lstm_params(vocab_size, num_hiddens, device):
...
+    def three():
+        return (normal((num_inputs, num_hiddens)),
+                normal((num_hiddens, num_hiddens)),
+                tf.Variable(tf.zeros(num_hiddens), dtype=tf.float32))
```
Comment (terrytangyuan) [inline chapter_recurrent-modern/lstm.md:226]: Add a newline

**Review by terrytangyuan — APPROVED** [review]

**Comment by mli** [comment]
Job d2l-en/PR-1683/2 is complete. 
Check the results at http://preview.d2l.ai/d2l-en/PR-1683/

**Pull request merged**